core/src/components/cat-button/cat-button.scss

```diff
@@ -78,6 +78,32 @@ $button-sizes: (
   }
 }
 
+// ----- group button
+
+.cat-group-button {
+  &-first {
+    border-top-right-radius: 0;
+    border-bottom-right-radius: 0;
+  }
+
+  &-middle {
+    border-radius: 0;
+  }
+
+  &-last {
+    border-top-left-radius: 0;
+    border-bottom-left-radius: 0;
+  }
+
+  &:focus-visible {
```

Comment:
shouldn't this be after the &:hover rule? otherwise if there is focus and we hover, it will be back to 1